src/openssl_wrap.cpp

```diff
@@ -1243,5 +1243,39 @@ EC_KEY *_EVP_PKEY_get0_EC_KEY(EVP_PKEY *pkey)
 {
     return OpensslCallPtr::callChecked(lib::OpenSSLLib::SSL_EVP_PKEY_get0_EC_KEY,pkey);
 }
+
+void _PKCS5_PBKDF2_HMAC(const char *pass, int passlen, const unsigned char *salt, int saltlen, int iter,
+                        const EVP_MD *digest, int keylen, unsigned char *out)
+{
+    OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_PKCS5_PBKDF2_HMAC, pass, passlen, salt, saltlen, iter, digest,
+                                  keylen, out);
+}
+
+void _ECDH_KDF_X9_63(unsigned char *out, size_t outlen, const unsigned char *Z, size_t Zlen, const unsigned char *sinfo,
+                     size_t sinfolen, const EVP_MD *md)
+{
+    OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_ECDH_KDF_X9_63, out, outlen, Z, Zlen, sinfo, sinfolen, md);
+}
+
+void _HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int key_len, const EVP_MD *md, ENGINE *impl)
+{
+    OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_HMAC_Init_ex, ctx, key, key_len, md, impl);
+}
+
+void _HMAC_Final(HMAC_CTX *ctx, unsigned char *md, unsigned int *len)
+{
+    OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_HMAC_Final, ctx, md ,len);
+}
+
+void _HMAC_Update(HMAC_CTX *ctx, const unsigned char *data, int len)
+{
+    OpensslCallIsOne::callChecked(lib::OpenSSLLib::SSL_HMAC_Update, ctx, data, len);
+}
+
+SSL_HMAC_CTX_Ptr _HMAC_CTX_new()
+{
+    return SSL_HMAC_CTX_Ptr{OpensslCallPtr::callChecked(lib::OpenSSLLib::SSL_HMAC_CTX_new)};
```

Comment:
There is some infrastructure with the createOpenSSLObject and createManagedOpenSSLObject function(templates). It unfortunately doesn't work for everything but should be imho the preferred pattern in this library.

Reply:
Done